examples/adjoints/adjoints_ex7/SigmaPhysics.C

```diff
@@ -0,0 +1,194 @@
+// The libMesh Finite Element Library.
+// Copyright (C) 2002-2020 Benjamin S. Kirk, John W. Peterson, Roy H. Stogner
+
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+
+
+
+#include "libmesh/getpot.h"
+
+#include "SigmaPhysics.h"
+
+#include "libmesh/fe_base.h"
+#include "libmesh/fe_interface.h"
+#include "libmesh/fem_context.h"
+#include "libmesh/mesh.h"
+#include "libmesh/quadrature.h"
+#include "libmesh/string_to_enum.h"
+#include "libmesh/system.h"
+#include "libmesh/equation_systems.h"
+#include "libmesh/zero_function.h"
+#include "libmesh/const_function.h"
+#include "libmesh/dirichlet_boundaries.h"
+#include "libmesh/dof_map.h"
+#include "libmesh/numeric_vector.h"
+#include "heatsystem.h"
+
+void SigmaPhysics::init_data (System & sys)
+{
+  T_var = dynamic_cast<FEMSystem&>(sys).add_variable ("T", static_cast<Order>(1), LAGRANGE);
+
+  GetPot infile("heat.in");
+  _k = infile("k", 1.0);
+  _analytic_jacobians = infile("analytic_jacobians", true);
+
+  // The temperature is evolving, with a first order time derivative
+  this->time_evolving(T_var, 1);
+
+  const boundary_id_type all_ids[4] = {0, 1, 2, 3};
+  std::set<boundary_id_type> all_bdys(all_ids, all_ids+(2*2));
+
+  std::vector<unsigned int> T_only(1, T_var);
+
+  ZeroFunction<Number> zero;
+
+  ConstFunction<Number> one(1.0);
+
+  // Most DirichletBoundary users will want to supply a "locally
+  // indexed" functor
+  sys.get_dof_map().add_dirichlet_boundary(DirichletBoundary (all_bdys, T_only, &one));
+
+  // Set adjoint boundary condition z(1,y) = y * (1 - y)
```

Comment:
Huh?  Looks like "z(partial Omega) = 0" to me.

Reply:
Sorry, that needs to be updates.